decoder/vaapidecoder_mpeg2.cpp

```diff
@@ -517,65 +517,18 @@ YamiStatus VaapiDecoderMPEG2::fillConfigBuffer()
         return status;
     }
 
-    m_configBuffer.width
+    uint32_t width
         = (m_sequenceExtension->horizontal_size_extension & 0x3)
           | (m_sequenceHeader->horizontal_size_value & 0xFFF);
-    m_configBuffer.height = (m_sequenceExtension->vertical_size_extension & 0x3)
+    uint32_t height = (m_sequenceExtension->vertical_size_extension & 0x3)
                             | (m_sequenceHeader->vertical_size_value & 0xFFF);
 
-    if (m_VAStart) {
-        // sequence start extension code with VA started has to conduct a
-        // port reconfiguration when stream size changes
-        if (m_configBuffer.width > m_videoFormatInfo.width
-            || m_configBuffer.height > m_videoFormatInfo.height) {
-            // need to re-start VA to generate new surfaces
-            status = VaapiDecoderBase::terminateVA();
-            if (status != YAMI_SUCCESS) {
-                return status;
-            }
-            m_VAStart = false;
-        }
-        else if (m_configBuffer.width != m_videoFormatInfo.width
-                 || m_configBuffer.height != m_videoFormatInfo.height) {
-            // VA surfaces can be re-used and client is notified
-            // public member from base class
-            m_videoFormatInfo.width = m_configBuffer.width;
-            m_videoFormatInfo.height = m_configBuffer.height;
-            m_videoFormatInfo.surfaceWidth = m_configBuffer.width;
-            m_videoFormatInfo.surfaceHeight = m_configBuffer.height;
-            m_configBuffer.surfaceWidth = m_configBuffer.width;
-            m_configBuffer.surfaceHeight = m_configBuffer.height;
-            status = YAMI_DECODE_FORMAT_CHANGE;
-        }
+    if (setFormat(width, height, width, height, kMinSurfaces + 1)) {
+        m_VAStart = false;
+        return YAMI_DECODE_FORMAT_CHANGE;
```

Comment:
This checked by isSurfaceGeometryChanged. Here is basic logical
1. Format change return for any width, height, susrfaceWidth, surfaceHeight, fourcc, pdb size change. This give user a chance to reconfig the renderer
2. We need flush dpb and output all previous frames since format changed.
3. We only reallocate surface pool when surface geometry changed (this make sure by decoder/vaapidecoder_base.cpp:L328)
Do you think this make sense?